libs/output-mapping/tests/Writer/TableDefinitionTest.php

```diff
@@ -53,27 +65,115 @@ public function configProvider(): \Generator
+            ],
         ];
+
+        // test native snowflake types
         yield [
             [
                 'source' => 'tableDefinition.csv',
                 'destination' => 'out.c-output-mapping.tableDefinition',
-                'columns' => ['Id', 'Name', 'birthday', 'created'],
+                'columns' => ['Id', 'Name', 'birthweight', 'created'],
+                'metadata' => [
+                    [
+                        'key' => TableDefinitionFactory::NATIVE_TYPE_METADATA_KEY,
+                        'value' => 'snowflake',
+                    ],
+                ],
+                'column_metadata' => [
+                    'Id' => (new Snowflake(
+                        Snowflake::TYPE_INTEGER,
+                        ['nullable' => false]
+                    ))->toMetadata(),
+                    'Name' => (new Snowflake(
+                        Snowflake::TYPE_TEXT,
+                        ['length' => '17', 'nullable' => false]
+                    ))->toMetadata(),
+                    'birthweight' => (new Snowflake(Snowflake::TYPE_DECIMAL, ['length' => '10,2']))->toMetadata(),
+                    'created' => (new Snowflake(Snowflake::TYPE_TIMESTAMP_TZ))->toMetadata(),
+                ],
+                'primary_key' => ['Id', 'Name'],
+            ],
+            true,
+            [
+                // INTEGER is an alias of NUMBER in snflk and describe returns the root type
+                'Id' => [
+                    'type' => Snowflake::TYPE_NUMBER,
+                    'length' => '38,0', // default integer length
+                    'nullable' => false,
+                ],
+                // LIKEWISE TEXT is an alias of VARCHAR
+                'Name' => [
+                    'type' => Snowflake::TYPE_VARCHAR,
+                    'length' => '17',
+                    'nullable' => false,
+                ],
+                'birthweight' => [
+                    'type' => Snowflake::TYPE_NUMBER,
+                    'length' => '10,2',
+                    'nullable' => true,
+                ],
+                // this one stays the same because it is not an alias
+                'created' => [
+                    'type' => Snowflake::TYPE_TIMESTAMP_TZ,
+                    'length' => '9', // timestamp_tz has length 9 apparently
+                    'nullable' => true,
+                ],
+            ],
+        ];
+        yield [
+            [
+                'source' => 'tableDefinition.csv',
+                'destination' => 'out.c-output-mapping.tableDefinition',
+                'columns' => ['Id', 'Name', 'birthweight', 'created'],
                 'column_metadata' => [],
                 'primary_key' => ['Id', 'Name'],
             ],
             false,
+            [],
         ];
     }
+
+    private static function assertDataType($metadata, $expectedType): void
+    {
+        $typeFound = false;
```

Comment:
can we split this method? because it does assert that the data type is same, but then also sometimes asserts that the length and nullable are same, and that *sometimes* is bugging me